docs/EXTENSIONS_GUIDE.md

```diff
@@ -0,0 +1,53 @@
+# Lisk Hub Extension Guideline
+``` ⚠️ Extension System is only avaliable in development mode ⚠️```
```

Comment:
What does it mean "development mode"? IMO the explanation should be more explicit about `npm run dev` vs. `npm run start` vs. `npm run build` vs. `npm run dist` vs. `npm run pack`